relays/ethereum/src/sync_loop.rs

```diff
@@ -131,7 +126,7 @@ pub fn run<P: HeadersSyncPipeline>(
 		let mut stall_countdown = None;
 		let mut last_update_time = Instant::now();
 
-		let mut source_maybe_client = None;
+		let mut source_client_is_online = false;
```

Comment:
I think that concept of we-can-execute-at-most-1-request-at-a-time is lost here. Before this PR, if you have e.g. asked Ethereum node about best header, you was unable to ask it for some header until best header has been received. In your concept, this flag has changed its meaning - it is `false` only when transport error has been reported by the underlying client (OK - and initially : ) ). I think that's the reason behind errors like this:
```
Substrate: [2020-06-19 15:26:59] 2020-06-19 15:26:59.941 http.worker30 DEBUG txpool  [0x31d7836d46b14977db3012267046897bbc13307fa78f8ce4d41c98ac61626b6b] Error importing: TooLowPriority { old: 10020844, new: 10020844 }

Substrate: [2020-06-19 15:26:59] 2020-06-19 15:26:59.941 http.worker30 DEBUG rpc  Response: {"jsonrpc":"2.0","error":{"code":1014,"message":"Priority is too low: (10020844 vs 10020844)","data":"The transaction has too low priority to replace another transaction already in the pool."},"id":330}.
```

I.e. we're spawning future1 that submits `headers[1..16]` to substrate node and then immediately spawning future2 that submits `headers[17..32]`. And there's a race, because both futures are asking Substrate node about account' nonce => two transactions are receiving the same nonce => submit fails.

Let's revert to previous behavior here (1 future at a time). There may be some other problems with this as well - i.e. when future that asks for best block is polled/completes after future that asks for best block, ...